### ISSUE TITLE
Figured out how to do it in eclipse,but do we have to learn git as well? 

### ISSUE THREAD

**Issue by karlnicholas** [CS111B-META-SUMMER-2016/CS111B-SAND-BOX#1]
Knowing "git" means doing things from the command line. Sure, it's a good idea. However, I've only done basic things from the command line. I haven't gone through a full merge with only command line operations. Eclipse have a "synchronize workspace" perspective that makes it much easier, once you learn how to do it. I'd probably say that learning source control from either direction (command line or visually) is different enough that it's basically something you're going to have to do separately.

**Comment by mdasc2** [comment]
Mmm, I was primarily asking for in the workforce or for future reference do most people/companies use commandline only or something like eclipse to upload to github.

**Comment by vinicius5581** [comment]
Do you guys understand the concept of branches? Perhaps it might be a better idea to have different branches and send pull requests in order to avoid conflicts.

If not make sure to pull before you push. That way you will always have the latest before you send your local changes.

**Comment by mdasc2** [comment]
I get branches as in you make separate branches in order to edit the file separately and merge it to the master branch after you are done.

What I'm wondering is that is it necessary to download git? Or can we do everything from eclipse.
I'm trying to figure out how to upload from eclipse to git without cloning a repo on github first like karl's guide showed how to do.

**Comment by karlnicholas** [comment]
I would say both. There are one or two things I haven't figured how to do in Egit (git for eclipse). It may depend on company culture, and they may not be using git. Also there may be automated or scripted pushes that are run from the command line. As a developer your first goal is understanding local vs remote repositories, branches, merging and some of the mundane and trouble shooting. Just using for any project is half the battle.

Sent from my iPhone

On Jun 27, 2016, at 8:44 PM, mdasc2 <notifications@github.com> wrote:

Mmm, I was primarily asking for in the workforce or for future reference do most people/companies use commandline only or something like eclipse to upload to github.

**Comment by karlnicholas** [comment]
You should always connect to a remote repo, get the latest, make your changes, then push them back. Even if you make something separately git will make you do that before it will let you push.

Sent from my iPhone

On Jun 28, 2016, at 3:08 AM, mdasc2 <notifications@github.com> wrote:

I get branches as in you make separate branches in order to edit the file separately and merge it to the master branch after you are done.

What I'm wondering is that is it necessary to download git? Or can we do everything from eclipse.
I'm trying to figure out how to upload from eclipse to git without cloning a repo on github first like karl's guide showed how to do.

**Comment by mdasc2** [comment]
Yeah, I see I had to connect and clone to the remote repo on github before I could make any changes. I couldn't just push a file straight into it.

**Comment by karlnicholas** [comment]
Kind of makes sense though, doesn’t it. You could be trying to push into something that already exists.

**Comment by vinicius5581** [comment]
Another way to get started is to do a git init on your local directory (repository) and then setup (connect) to the remote (origin) and then fetch.

```
git init
git remote add origin $url_of_clone_source
git fetch origin
git checkout -b master --track origin/master # origin/master is clone's default
```

Then you can reset the tree to get the commit you want:

`git reset origin/master #` or whatever commit you think is proper...

source: http://stackoverflow.com/questions/5377960/whats-the-best-practice-to-git-clone-into-an-existing-folder

**Comment by karlnicholas** [comment]
git checkout -b master --track origin/master # origin/master is clone's default

I think this will checkout the entire project? Perhaps it's better to do a `git pull` at this point just to get a copy of everything and checkout what you work on? I'm not a git command-line guru.

**Comment by vinicius5581** [comment]
On the checkout command you are basically syncing the branches. Your local master branch to track the origin/master branch on this case. 

`Branch master set up to track remote branch master from origin.`

If you try to pull at that point you would get an error:

```
There is no tracking information for the current branch.
Please specify which branch you want to merge with.
See git-pull(1) for details.

    git pull <remote> <branch>

If you wish to set tracking information for this branch you can do so with:

    git branch --set-upstream-to=origin/<branch> master
```
